VideoWeb/VideoWeb/ClientApp/src/app/services/security/auth.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import { OidcSecurityService } from 'angular-auth-oidc-client';
+import { ReplaySubject, Observable } from 'rxjs';
+import { first, switchMap, tap } from 'rxjs/operators';
+
+@Injectable()
+export class AuthService {
+    private checkAuthCompleted$ = new ReplaySubject(1);
+
+    constructor(private oidcSecurityService: OidcSecurityService) { }
+
+    public get isAuthenticated$(): Observable<boolean> {
+        return this.checkAuthCompleted$.pipe(
+            first(),
+            switchMap((_) => this.oidcSecurityService.isAuthenticated$)
+        );
+    }
+
+    public checkAuth(): Observable<boolean> {
+        return this.oidcSecurityService.checkAuth().pipe(tap((_) => this.checkAuthCompleted$.next()));
```

Comment:
Looks like this just emits the checkAuthCompleted$ subject when checkAuth emits. Could we just return a reference to the checkAuth observable?

Reply:
Instead of adding another observable to subscibe to, I mean.

Reply:
This is to make ignore values from isAuthenticated$ before checkAuth() emits a value, I don't have idead to just return a reference, I think create an another observable to subscibe is necessary